packages/block-library/src/image/edit.js

```diff
@@ -87,80 +87,73 @@ const isTemporaryImage = ( id, url ) => ! id && isBlobURL( url );
  */
 const isExternalImage = ( id, url ) => url && ! id && ! isBlobURL( url );
 
-export class ImageEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.updateAlt = this.updateAlt.bind( this );
-		this.updateAlignment = this.updateAlignment.bind( this );
-		this.onFocusCaption = this.onFocusCaption.bind( this );
-		this.onImageClick = this.onImageClick.bind( this );
-		this.onSelectImage = this.onSelectImage.bind( this );
-		this.onSelectURL = this.onSelectURL.bind( this );
-		this.updateImage = this.updateImage.bind( this );
-		this.onSetHref = this.onSetHref.bind( this );
-		this.onSetTitle = this.onSetTitle.bind( this );
-		this.getFilename = this.getFilename.bind( this );
-		this.onUploadError = this.onUploadError.bind( this );
-		this.onImageError = this.onImageError.bind( this );
-
-		this.state = {
-			captionFocused: false,
-		};
+function getFilename( url ) {
+	const path = getPath( url );
+	if ( path ) {
+		return last( path.split( '/' ) );
 	}
+}
 
-	componentDidMount() {
-		const { attributes, mediaUpload, noticeOperations } = this.props;
-		const { id, url = '' } = attributes;
+export function ImageEdit( {
+	attributes: {
+		url = '',
+		alt,
+		caption,
+		align,
+		id,
+		href,
+		rel,
+		linkClass,
+		linkDestination,
+		title,
+		width,
+		height,
+		linkTarget,
+		sizeSlug,
+	},
+	setAttributes,
+	isSelected,
+	className,
+	noticeUI,
+	insertBlocksAfter,
+	noticeOperations,
+	onReplace,
+} ) {
+	const selected = useSelect( ( select ) => {
+		const { getMedia } = select( 'core' );
+		const { getSettings } = select( 'core/block-editor' );
+		const { mediaUpload, imageSizes, isRTL, maxWidth } = getSettings();
+		const image = id && isSelected ? getMedia( id ) : null;
 
-		if ( isTemporaryImage( id, url ) ) {
-			const file = getBlobByURL( url );
-
-			if ( file ) {
-				mediaUpload( {
-					filesList: [ file ],
-					onFileChange: ( [ image ] ) => {
-						this.onSelectImage( image );
-					},
-					allowedTypes: ALLOWED_MEDIA_TYPES,
-					onError: ( message ) => {
-						noticeOperations.createErrorNotice( message );
-					},
-				} );
-			}
-		}
+		return {
+			image,
+			maxWidth,
+			isRTL,
+			imageSizes,
+			mediaUpload,
+		};
+	} );
```

Comment:
You need to add a dependency array here. (And I really wish we had an ESLint rule for this.)

I think the dependency array should be: `[ id, isSelected ]`.